src/TgSharp.Core/Session.cs

```diff
@@ -90,23 +162,30 @@ private static int CurrentTime ()
 #endif
 
         public string SessionUserId { get; set; }
-        internal DataCenter DataCenter { get; set; }
+        public DataCenter DataCenter { get; set; }
         public AuthKey AuthKey { get; set; }
         public ulong Id { get; set; }
         public ulong Salt { get; set; }
         public int TimeOffset { get; set; }
         public long LastMessageId { get; set; }
         public int SessionExpires { get; set; }
-        public TLUser TLUser { get; set; }
-        private Random random;
-
-        private ISessionStore store;
 
-        public Session(ISessionStore store)
-        {
-            random = new Random();
-            this.store = store;
+        public int UserId { get; set; }
+        private TLUser user = null;
+        public TLUser User {
+            get { return user; }
+            internal set {
+                user = value;
+                if (value != null) {
+                    UserId = value.Id;
+                }
+            }
         }
+        public bool AuthenticatedSuccessfully {
```

Comment:
Technically we don't have to save UserId at all because we can just run GetSelfUser and If we're not authenticated we get an error